### ISSUE TITLE
Hyperparameters have not been explored.

### ISSUE THREAD

**Issue by bestksw** [polyaxon/polyaxon#595]
### Describe the bug
I'm newbie of Polyaxon.
I've followed this docs https://docs.polyaxon.com/concepts/quick-start-internal-repo/.
All experiments described in docs work, but when I run polyaxonfile_hyperparams_bo.yml, one of example polyaxonfile, I found that the hyper parameters are never changed.

I send the command to shell like this.
```
$ polyaxon run -f polyaxonfile_hyperparams_bo.yml 
Creating an experiment group with the following definition:
----------------  ------------------
Search algorithm  bo
Concurrency       40 concurrent runs
Early stopping    deactivated
----------------  ------------------
Experiment group 35 was created
```

and then, I checked the result.
```
$ polyaxon group -g 35 experiments -p

Experiments for experiment group `35`.


Navigation:

-----  --------
next   --page=2
count  43
-----  --------

Experiments:

  id  unique_name                      dropout    num_steps  activation      batch_size    num_epochs    learning_rate
----  -----------------------------  ---------  -----------  ------------  ------------  ------------  ---------------
1344  testexp.quick-start.35.1344       0.25          100  relu                   128             1            0.001
1342  testexp.quick-start.35.1342       0.25          100  relu                   128             1            0.001
1343  testexp.quick-start.35.1343       0.25          100  relu                   128             1            0.001
1327  testexp.quick-start.35.1327       0.25          100  relu                   128             1            0.001
1339  testexp.quick-start.35.1339       0.25          100  relu                   128             1            0.001
1337  testexp.quick-start.35.1337       0.25          100  relu                   128             1            0.001
1341  testexp.quick-start.35.1341       0.25          100  relu                   128             1            0.001
1331  testexp.quick-start.35.1331       0.25          100  relu                   128             1            0.001
1325  testexp.quick-start.35.1325       0.25          100  relu                   128             1            0.001
1318  testexp.quick-start.35.1318       0.25          100  relu                   128             1            0.001
1324  testexp.quick-start.35.1324       0.25          100  relu                   128             1            0.001
1322  testexp.quick-start.35.1322       0.25          100  relu                   128             1            0.001
1329  testexp.quick-start.35.1329       0.25          100  relu                   128             1            0.001
1336  testexp.quick-start.35.1336       0.25          100  relu                   128             1            0.001
1319  testexp.quick-start.35.1319       0.25          100  relu                   128             1            0.001
1334  testexp.quick-start.35.1334       0.25          100  relu                   128             1            0.001
1321  testexp.quick-start.35.1321       0.25          100  relu                   128             1            0.001
1326  testexp.quick-start.35.1326       0.25          100  relu                   128             1            0.001
1328  testexp.quick-start.35.1328       0.25          100  relu                   128             1            0.001
1330  testexp.quick-start.35.1330       0.25          100  relu                   128             1            0.001
```
As you can see, the hyperparameters have not been explored during the experiments. 

### To Reproduce

```
$ git clone https://github.com/polyaxon/polyaxon-quick-start
$ cd polyaxon-quick-start
$ polyaxon run -f polyaxonfile_hyperparams_bo.yml
$ polyaxon group -g 35 experiments -p
```

### Expected behavior
The hyperparameters shoud have been explored during the experiments to find best hyperparameters.

### Environment
Rancher 2.0 with its own privided custom cluster in Ubuntu 18.04 Server.
Polyaxon server v0.5.6 installed by helm
Polyaxon client v0.5.6 executed in Ubuntu 16.04 Desktop

**Comment by bestksw** [comment]
When I remove learning_rate, dropout, activation from inputs like below, the hyperparameters have been exploared finally.
If hyperparameters are mixing in inputs and hptuning.matrix, I guess it makes bugs.
```
---
version: 1

kind: group

framework: tensorflow

inputs:
  - name: batch_size
    is_optional: true
    default: 128
    type: int
  - name: num_steps
    is_optional: true
    default: 100
    type: int
  - name: num_epochs
    is_optional: true
    default: 1
    type: int

hptuning:
  concurrency: 40
  bo:
    n_iterations: 15
    n_initial_trials: 40
    metric:
      name: loss
      optimization: minimize
    utility_function:
      n_warmup: 5
      n_iter: 10
      acquisition_function: ucb
      kappa: 2.576
      gaussian_process:
        kernel: matern
        length_scale: 1.0
        nu: 1.9
        n_restarts_optimizer: 0

  matrix:
    learning_rate:
      uniform: [0.001, 0.01]
    dropout:
      values: [0.2, 0.22, 0.25, 0.3, 0.35]
    activation:
      values: [relu, sigmoid]

build:
  image: tensorflow/tensorflow:1.4.1-py3
  build_steps:
    - pip3 install --no-cache-dir -U polyaxon-client[gcs]==0.5.5

run:
  cmd:  python3 model.py --batch_size={{ batch_size }} \
                         --num_steps={{ num_steps }} \
                         --learning_rate={{ learning_rate }} \
                         --dropout={{ dropout }} \
                         --num_epochs={{ num_epochs }} \
                         --activation={{ activation }}
```
